src/main/java/graphql/schema/idl/SchemaPrinter.java

```diff
@@ -365,17 +369,17 @@ public String print(GraphQLSchema schema) {
         return result;
     }
 
-    private interface TypePrinter<T> {
+    private interface SchemaElementPrinter<T> {
 
-        void print(PrintWriter out, T type, GraphqlFieldVisibility visibility);
+        void print(PrintWriter out, T schemaElement, GraphqlFieldVisibility visibility);
 
     }
 
     private boolean isIntrospectionType(GraphQLNamedType type) {
         return !options.isIncludeIntrospectionTypes() && type.getName().startsWith("__");
     }
 
-    private TypePrinter<GraphQLScalarType> scalarPrinter() {
+    private SchemaElementPrinter<GraphQLScalarType> scalarPrinter() {
```

Comment:
SchemaElementPrinter is more correct because we top level types and GraphQLDirectives to be printed